lib/puppet/provider/package/rpm.rb

```diff
@@ -310,7 +311,9 @@ def rpmvercmp(str1, str2)
   def insync?(is)
     return false if [:purged, :absent].include?(is)
     should = resource[:ensure]
-    0 == rpm_compareEVR(rpm_parse_evr(should), rpm_parse_evr(is))
+    is.split(self.class::MULTIVERSION_SEPARATOR).any? do |version|
```

Comment:
I don't know enough about RPM to make a good judgment: but we should consider the ramifications of using `.any?` here. I would expect using something like `ensure => 1.1.1` with only one version would mean that the system would remove any other versions installed. If that was not the case with `ensure` alone, we would need a new property in the package type and provider to let a user specify they only want a single version of the package present.

That situation is a primary concern with this PR: how would a user write a package resource to specify they only want one version of the package installed?

Reply:
It's possible that multiple packages are so rare this isn't really an issue, but I don't know enough about users or RPM to know if that's the case.

Reply:
There are specific packages (can be specified as list in `yum.conf`, `installonlypkgs`, by default includes kernel packages: [https://linux.die.net/man/5/yum.conf]) for which yum upgrade/downgrade keeps also the previous version (until installonly_limit is reached)

Based on your comment I think we should keep by default one version behavior and add a package/provider property for multiversion support (`installonlypkg`) and enable multiversion support only if the property is set.

Reply:
I think having a new property sounds like a good approach.